### ISSUE TITLE
flex 2.6.3 breaks compiling binutils.

### ISSUE THREAD

**Issue by hussamT** [westes/flex#154]
I get the following error:
make[2]: Entering directory '/home/hussam/cache/binutils/src/binutils-build/binutils'
gcc -march=x86-64 -mtune=native -O2 -pipe -fstack-protector-strong -Wl,-O1,--sort-common,--as-needed,-z,relro -o sysinfo sysinfo.o syslex_wrap.o
syslex_wrap.o: In function `yylex':
syslex_wrap.c:(.text+0xc63): undefined reference to `yywrap'
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:1370: sysinfo] Error 1
make[2]: Leaving directory '/home/hussam/cache/binutils/src/binutils-build/binutils'
make[1]: *** [Makefile:3612: all-binutils] Error 2
make[1]: Leaving directory '/home/hussam/cache/binutils/src/binutils-build'

Reverting to 2.6.1 fixes it.

**Comment by westes** [comment]
What about 2.6.2?

**Comment by hussamT** [comment]
2.6.2 lets binutils compile correctly.

**Comment by nvinson** [comment]
@hussamT Could you state which version of binutils you're trying to build?  I am not able to repeat this issue with binutils-2.25.1, binutils-2.26, or binutils-2.27.  I also have not seen it in Gentoo's tracker for this issue (https://bugs.gentoo.org/show_bug.cgi?id=598186)

**Comment by hussamT** [comment]
Hi nvinsion. I compile binutils from 2.27 branch.
https://sourceware.org/git/?p=binutils-gdb.git;a=shortlog;h=refs/heads/binutils-2_27-branch

**Comment by nvinson** [comment]
Thanks.  I was able to reproduce it now.

In syslex.l I see
```
#ifndef yywrap
static int yywrap (void) { return 1; }
#endif
```
Which will prevent yywrap from being defined in flex-2.6.3.  The source file syslex.c is not regenerated when building from the binutils-2.27 release tarball, which is why I didn't see it in my attempts to reproduce.

**Comment by jgalar** [comment]
Building Babeltrace 1.5.x fails with a similar error:
https://gist.github.com/jgalar/a4c447e8291264e5d8acc0a3165d2e14

**Comment by westes** [comment]
Looks like the same issue as #162; please confirm that you see this fixed for you on master now.

**Comment by hussamT** [comment]
I can compile binutils again using flex from git master branch. Thank you

**Comment by jgalar** [comment]
Just wondering if there is an ETA available for `2.6.4`?

Arch Linux (maybe others?) is distributing `2.6.3` which is forcing users to downgrade `flex` in order to build a number of packages.

**Comment by mrcmunir** [comment]
@jgalar If use Arch linu use flex-git from aur  https://aur.archlinux.org/packages/flex-git/
